### PR TITLE
Feature: Operator UI + APIs

### LINKED ISSUE #349: Add one "OPERATOR" user type in db
### What is this issue related to?

Feature

### Description

 We have currently two type of users in database, the **CUSTOMER** can only create keys and use APIs and **ADMIN** can add images to the database.
Now we need to **implement** a feature in which an **OPERATOR** role user type which will handle the queries asked by users, view them, respond or track it.

### Expected Behaviour
- When operator login with user type operator, it will be redirecting to operator dashboard.
- It will be viewing the queries in a view, which will look similar to diagram attached.
- It can check for **Active** and **Archive** via activityStatus query params.
- If activityStatus is true then it will fetch **Active** data and **Archive** in false.
- In the layout there will a small card type of component which will show user name and few queries with trimmed message and full view on send response button.
- There will be pagination to make the UI performant by fetching one page with some limit at one request.

image

### Current Behaviour
- We don't have any flow for operator in the app. Need to implement like above mentioned.

### Aligned task with folks:
- @Ayushsanjdev - Writing controller and services to get contact-us data based on activityStatus and make a common api for pagination and writing test cases + help with UI for the same to Sujal.
- @asharma991 - Writing controller and services to respond to the users by updating the reply key in collection and writing test cases + if required do some UI task.
- @sujal111 - Making mostly the UI part of the flow like tabs, cards with response modal all together with pagination component.

### Confirmation Checklist

- [x] I have searched for similar issues before creating this one.
- [x] I have provided all the necessary information to understand and reproduce the issue.

### PR DESCRIPTION
**Pull request by asharma991**
### Summary
#349 

Add new route to revert to the customer.
Add new route to common/pagination to get customer queries.

### Description

Adding new route to revert to the customer using new flow for the Contact Us.
Add new route to common/pagination to get queries of the customer in paginated form for the ContactUs.
This is a part of #349.
New controllers and services added have been tested thoroughly and verified in the API client.

### How to Test the Changes

Test the ```/api/operator/revert``` route by using the sample payload and the cookie header with the jwt.

**Sample Payload**:
```bash
curl --location --request PUT 'http://localhost:5000/api/operator/revert' \
--header 'Cookie: jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJkYXRhIjp7ImZpcnN0TmFtZSI6IkF5dXNoIiwibGFzdE5hbWUiOiJTYW5qIiwiZW1haWwiOiJheXVzaHNhbmpwcm9AZ21haWwuY29tIiwidXNlcklkIjoiNjZiOGY2NTExODhlYWI2NzMyNzkwMzU1IiwidXNlclR5cGUiOiJPUEVSQVRPUiJ9LCJpYXQiOjE3MjYxNjY5NzksImV4cCI6MTcyNjI1MzM3OX0.8THsjLg4n0YL-mBKpwB5zG90Fv2RXRlQsjHDzMk7A2c' \
--header 'Content-Type: application/json' \
--data '{
    "id":"66bf3695a4e3b9dab1d6f8e9",
    "reply": "Hi there where are you now"
}'
```

**Happy Response**:
```json
{
    "statusCode": 200,
    "message": "Response sent to customer successfully"
}
```

**Already Replied**
```json
{
    "statusCode": 409,
    "error": "Conflict",
    "message": "Already sent the response for this query!"
}
```

## pagination
Test the ```/api/common/pagination``` route by using the sample queries and the cookie header with the jwt.

* Sample query params:

```
curl --location 'localhost:5000/api/common/pagination?type=queries&page=1&limit=2&active=true' \
--header 'Cookie: jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJkYXRhIjp7ImZpcnN0TmFtZSI6IkF5dXNoIiwibGFzdE5hbWUiOiJTYW5qIiwiZW1haWwiOiJheXVzaHNhbmpwcm9AZ21haWwuY29tIiwidXNlcklkIjoiNjZiOGY2NTExODhlYWI2NzMyNzkwMzU1IiwidXNlclR5cGUiOiJPUEVSQVRPUiJ9LCJpYXQiOjE3MjYxNjY5NzksImV4cCI6MTcyNjI1MzM3OX0.8THsjLg4n0YL-mBKpwB5zG90Fv2RXRlQsjHDzMk7A2c'
```
Happy response:
```
{
    "message": "Successful",
    "statusCode": 200,
    "total": 8,
    "pages": 4,
    "results": [
        {
            "_id": "66b8f98b188eab673279035f",
            "message": "Hi There I want some very nice logos",
            "activityStatus": true,
            "createdAt": "2024-08-11T17:48:59.075Z",
            "updatedAt": "2024-08-11T17:48:59.075Z"
        },
        {
            "_id": "66bf368da4e3b9dab1d6f8de",
            "message": "Contact us if you are experiencing issues with our product or have any questions",
            "activityStatus": true,
            "createdAt": "2024-08-16T11:22:53.581Z",
            "updatedAt": "2024-08-16T11:22:53.581Z"
        }
    ]
}
```

Params not filled:
```
{
    "message": "\"type\" is required",
    "statusCode": 422,
    "error": "Unprocessable Entity"
}
```

### Changes

A simple controller has been written for reverting and getting queries of customers + a new service for updating the form is implemented. Few mocks have been added to test the changes.

### Screenshots or Recordings (Optional)

* revert customer:
image
image

* get queries for operator:
image
image

* UI
image

image

## Checklist

- [x] I have tested the changes locally and they work as expected.
- [x] I have added/updated tests that cover the changes.
- [x] I have updated the documentation to reflect the changes.
- [x] This pull request follows the project's coding guidelines.

### CODE REVIEW

**Comment by Ayushsanjdev** [comment]
@asharma991 Please merge this in your dev(forked).
and close this PR, we will together raise one PR.

File: server/services/ContactUs.js [hunk 1]
```diff
@@ -37,7 +38,40 @@ async function createForm(formData) {
...
+/**
+ * Updates form in the DB
+ * @param {String} formId
+ * @param {String} email
+ * @param {String} reply
+ * @param {String} operatorId
+ */
+async function updateForm(formId, email, reply, operatorId) {
+  try {
+    const currentForm = await ContactUs.findOne({ _id: formId, email });
```
Comment (amankumarsingh77) [inline server/services/ContactUs.js]: Wouldn't it be better to use findById() as _id is an index by default? Also, _id is a unique identifier so I don't think we need email.

**Review by amankumarsingh77 — CHANGES_REQUESTED** [review]

File: server/services/ContactUs.js (hunk 1 above)
Comment (asharma991) [reply server/services/ContactUs.js]: email is removed in the latest implementation. I'll change it to findById

Comment (Ayushsanjdev) [reply server/services/ContactUs.js]: updated the changes: @amankumarsingh77 @asharma991

**Review by amankumarsingh77 — APPROVED** [review]

**Pull request merged**